### PR TITLE
Allow user to set grayscale when replacing videos (mp4/avi only) 

### PR DESCRIPTION
**Pull request by roomrys**
### Description
Show the same dialog for adding videos upon replacing videos (`ImportItemWidget`) which gives users the ability to select grayscale on mp4/avi videos -  no other import-able backends currently support grayscale.

The `ReplaceVideo` command continues to only work properly if the replacement video has the same extension as the current video - this limitation was not addressed in this PR.

### Types of changes

- [ ] Bugfix
- [x] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
None.

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/wiki/Developer-Guide) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

File: sleap/gui/commands.py
Comment (roomrys) [inline sleap/gui/commands.py:1586]: This bit of code will pop up for EVERY video that has labeled frames past the replacement video length - is it safe to assume that there will not be many cases where the above is true? How often do our users replace videos? How many videos do they replace at once? Otherwise, we will need to create a pop-up box similar to the `ImportItemWidget` which shows all problem videos in a single dialog.

File: sleap/gui/commands.py
Comment (roomrys) [inline sleap/gui/commands.py:1556]: We really shouldn't be changing the backend before getting approval from the user. To separate into `ask` and `do_action`, we could read the number of frames using cv2:
```py
reader = cv2.VideoCapture(new_path)
last_vid_frame = int(reader.get(cv2.CAP_PROP_FRAME_COUNT))
```

**Review by roomrys — COMMENTED** [review]
Ask permission before changing backend (even if it will be reset later on).

File: sleap/gui/commands.py
Comment (roomrys) [reply sleap/gui/commands.py]: The warning is now built into the `ImportItemWidget` that pops up to create our list of files to import.

File: sleap/gui/commands.py
Comment (roomrys) [reply sleap/gui/commands.py]: I chose to go with the above recommendation and create a new reader for each video to (possibly) import.

File: sleap/gui/commands.py
Comment (roomrys) [inline sleap/gui/commands.py:1554]: This line is only used for displaying information to user. Do we need it (i.e., is it worth any lag that it causes)?

File: sleap/gui/commands.py
Comment (roomrys) [inline sleap/gui/commands.py:1567]: Will a user ever import the same video more than once?

File: sleap/gui/commands.py
Comment (roomrys) [inline sleap/gui/commands.py:1576]: This is triggered even when labels are not added/changed (but another "change" was registered). Is there a way to check that just the labels object has changes?

**Review by roomrys — COMMENTED** [review]
Added a few questions/considerations for my reviewer.

File: sleap/gui/commands.py
Comment (talmo) [reply sleap/gui/commands.py:1554]: This is pretty minimal -- I'm ok with it in favor of being more verbose about potential catastrophic situations

File: sleap/gui/commands.py
Comment (talmo) [reply sleap/gui/commands.py:1567]: They shouldn't be able to!

File: sleap/gui/commands.py
Comment (talmo) [reply sleap/gui/commands.py:1576]: That would be amazing -- I've looked into it in the past but it doesn't seem trivial. Search for `"has_changes"` across the codebase.

In general, it's just a tricky pattern to implement. Here's some relevant literature: https://refactoring.guru/design-patterns/state

File: sleap/gui/dialogs/importvideos.py
Comment (talmo) [inline sleap/gui/dialogs/importvideos.py:158]: Nit: `... else ""` instead of `... else str()`

File: sleap/gui/dialogs/importvideos.py
Comment (talmo) [inline sleap/gui/dialogs/importvideos.py:558]: Really?? I would think you wouldn't need to override it, or at least have a `super().paint(...)` in there?

**Review by talmo — CHANGES_REQUESTED** [review]

File: sleap/gui/commands.py
Comment (roomrys) [reply sleap/gui/commands.py:1567]: @talmo I was able to. We can open an issue about this and fix it in another PR?

File: sleap/gui/dialogs/importvideos.py
Comment (roomrys) [reply sleap/gui/dialogs/importvideos.py:558]: Admittedly, I meant to test this myself and had not gotten to it - will test before next review.

File: sleap/gui/commands.py
Comment (roomrys) [reply sleap/gui/commands.py:1530]: I edited `MissingFilesDialog` to only allow the current video type using the `replace` kwarg, but also raise an error in `ReplaceVideo.do_action()`.

File: sleap/gui/dialogs/importvideos.py
Comment (roomrys) [reply sleap/gui/dialogs/importvideos.py:558]: It does not seem to be required by Qt.

File: sleap/gui/dialogs/importvideos.py
Comment (talmo) [inline sleap/gui/dialogs/importvideos.py:98]: Big no-no here! Mutable types should not be instantiated as function defaults! More info: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

See also the [typing module docs](https://docs.python.org/3.7/library/typing.html) for the proper way to typehint `dict`s in python 3.7.

What we want here is something like:
```
    def __init__(self, filenames: List[str], messages: Optional[Dict[str, str]] = None, *args, **kwargs):
        if messages is None:
            messages = dict()
```

File: sleap/gui/dialogs/importvideos.py
Comment (talmo) [inline sleap/gui/dialogs/importvideos.py:272]: Type hint should be `import_type: Dict[str, Any]` or something

**Review by talmo — CHANGES_REQUESTED** [review]
A couple more quick fixes and you're good to go

File: sleap/gui/dialogs/importvideos.py
Comment (roomrys) [reply sleap/gui/dialogs/importvideos.py]: Got it! Thanks

**Pull request merged**
